packages/core/src/browser/keybinding.ts

```diff
@@ -603,14 +591,15 @@ export class KeybindingRegistry {
         this.keyboardLayoutService.validateKeyCode(keyCode);
         this.keySequence.push(keyCode);
         const bindings = this.getKeybindingsForKeySequence(this.keySequence);
-
-        if (bindings.partial.length === 0 && this.tryKeybindingExecution(bindings.full, event)) {
+        const full = bindings.full.find(binding => this.isEnabled(binding, event));
+        const partial = bindings.partial.find(binding => this.isEnabled(binding, event));
+        if (full && full.scope !== undefined && partial && partial.scope !== undefined && full.scope >= partial.scope || full && !partial) {
 
             this.keySequence = [];
             this.statusBar.removeElement('keybinding-status');
 
-        } else if (bindings.partial.some(binding => this.isEnabled(binding, event))) {
-
+            this.executeKeyBinding(full, event);
+        } else if (partial !== undefined) {
```

Comment:
`if(partial)` partial is of an object type there is no equality gotchas for it

Reply:
fixed